src/person/person.controller.ts

```diff
@@ -96,9 +102,20 @@ export class Person {
     // check that 'directGroup' field exists
     if (!person.directGroup) {
       throw new Error('a person must have a direct group');
-    }
+    }  
     // delete empty or null field that are not necessary
     utils.filterEmptyField(person, ['rank', 'phone', 'mobilePhone', 'address', 'job']);
+    
+    // Chack some validation
+    // Check if personalNumber equal to identityCard
+    if (!(!person.personalNumber && !person.identityCard) && person.personalNumber === person.identityCard) {
```

Comment:
!(!A && !B) = A || B

Reply:
Changed condition  a & b  & a===b
Because only if both exist we need condition